### DISCUSSION TITLE
Error response from daemon: Get https://ghcr.io/v2/: denied

### DISCUSSION THREAD

**Discussion by srivatsamarichi** [community/community#26832] [Packages]
<p>Hi there, I’m trying to docker login to GHCR and I encounter this error. <a href="https://github.com/srivatsamarichi/piggymetrics/runs/1176830262?check_suite_focus=true" rel="noopener nofollow ugc">Here</a> is my workflow run. Can someone help me here? Thanks</p>

**Comment by airtower-luna** [comment]
<p>The login command as such looks correct. Can you check that your <code>CR_TOKEN</code> secret has the correct permissions, as described in <a href="https://docs.github.com/en/free-pro-team@latest/packages/managing-container-images-with-github-container-registry/pushing-and-pulling-docker-images#authenticating-to-github-container-registry">Authenticating to GitHub Container Registry</a>?</p>
<p>You can also use a command like the following from your <strong>local</strong> (I’m emphasizing local because a base64 encoded version of your token will be part of the output) machine to check if the token works. Replace <code>$USER</code> and <code>$CR_TOKEN</code> with your username and token, respectively, for the token preferably by setting an environment variable.</p>
<pre><code class="lang-bash">curl -v -u $USER:$CR_TOKEN https://ghcr.io/v2/
</code></pre>
<p>If you can authenticate with the token you’ll get a <code>200 OK</code> HTTP response, with a <code>Docker-Distribution-Api-Version: registry/2.0</code> header.</p>

**Comment by srivatsamarichi** [comment]
<p>Hi, I am able to successfully login using the CR_PAT generated locally on my machine. However, I run into this issue when initiated from the GitHub action workflow.</p>

**Comment by clarkbw** [comment, marked as answer]
<p>I’m not sure what’s wrong here if the CR_PAT works locally it should work this way.  Here are some ideas to try:</p>
<p>Reorder the command, docker login takes the server as the final argument but on linux it usually works fine. You could also use <code>${{ github.repository_owner }}</code> instead of your username but again this doesn’t usually cause errors.</p>
<p><code>echo ${{ secrets.CR_PAT }} | docker login -u ${{ github.repository_owner }} --password-stdin ghcr.io</code></p>
<p>Try using the <a href="https://github.com/docker/login-action#github-container-registry">login-action</a> which simplifies all of this.</p>
<pre><code class="lang-yaml">- name: Login to GitHub Container Registry
        uses: docker/login-action@v1
        with:
          registry: ghcr.io
          username: ${{ github.repository_owner }}
          password: ${{ secrets.CR_PAT }}
</code></pre>

**Reply by krebznet** [reply]
Yup that is exactly right! It works around some of lower level auth token scoping

**Comment by srivatsamarichi** [comment]
<p>works like a charm, thanks a lot.</p>
